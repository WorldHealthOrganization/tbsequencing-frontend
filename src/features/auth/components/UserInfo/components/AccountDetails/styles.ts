import { css } from '@emotion/react';

export const section = css({
  marginTop: 24,
  marginBottom: 24,
});

export const sectionInfo = css({
  marginTop: 8,
  fontSize: '16px',
});
