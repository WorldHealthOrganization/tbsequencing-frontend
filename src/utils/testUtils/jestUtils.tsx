import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
