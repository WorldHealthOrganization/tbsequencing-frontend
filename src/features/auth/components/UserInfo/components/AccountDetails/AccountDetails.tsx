/** @jsxImportSource @emotion/react */
import React, { useState } from 'react';
import { SerializedStyles } from '@emotion/react';
import { useMsal } from '@azure/msal-react';
import * as styles from './styles';
import AppPaper from '../../../../../../components/AppPaper';
import H2 from '../../../../../../components/typography/H2';
import H3 from '../../../../../../components/typography/H3';
import PrimaryText from '../../../../../../components/typography/PrimaryText';
import AppButton from '../../../../../../components/AppButton';
import ChangePasswordModal from './components/ChangePasswordModal';

interface Props {
  containerStyle: SerializedStyles;
}

export const AccountDetails = ({ containerStyle }: Props) => {
  const [changePasswordModalVisible, setChangePasswordModalVisible] = useState<boolean>(false);

  const { accounts } = useMsal();

  if (!accounts[0]) {
    return null;
  }

  const handleChangePasswordClick = () => {
    setChangePasswordModalVisible(true);
  };

  const onCloseModal = () => {
    setChangePasswordModalVisible(false);
  };

  return (
    <AppPaper style={containerStyle}>
      <H2>Account Details</H2>
      <section css={styles.section}>
        <H3>Email</H3>
        <PrimaryText style={styles.sectionInfo}>{accounts[0].username}</PrimaryText>
      </section>
      <section css={styles.section}>
        <H3>Password</H3>
        <PrimaryText style={styles.sectionInfo}>************************************</PrimaryText>
      </section>
      <section css={styles.section}>
        <AppButton onClick={handleChangePasswordClick} variant="outlined" size="large">Change password</AppButton>
      </section>
      <ChangePasswordModal handleClose={onCloseModal} isOpened={changePasswordModalVisible} />
    </AppPaper>
  );
};

export default AccountDetails;
