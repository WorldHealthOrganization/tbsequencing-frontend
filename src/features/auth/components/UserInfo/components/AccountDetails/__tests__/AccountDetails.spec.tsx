import React from 'react';
import { render, getTreeJSONWithRedux } from '../../../../../../../utils/testUtils/jestUtils';
import AccountDetails from '../AccountDetails';

describe('AccountDetails component test', () => {
  test('renders AccountDetails component without error', () => {
    render(
      <AccountDetails />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <AccountDetails />,
    );

    expect(tree).toMatchSnapshot();
  });
});
