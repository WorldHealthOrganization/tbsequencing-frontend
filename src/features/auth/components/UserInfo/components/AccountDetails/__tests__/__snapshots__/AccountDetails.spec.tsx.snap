// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDetails component test Should render correctly 1`] = `null`;
